packages/api/src/models/animalGroupModel.js

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import baseModel from './baseModel.js';
+
+class AnimalGroup extends baseModel {
+  static get tableName() {
+    return 'animal_group';
+  }
+
+  static get idColumn() {
+    return 'id';
+  }
+
+  // Optional JSON schema. This is not the database schema! Nothing is generated
+  // based on this. This is only used for validation. Whenever a model instance
+  // is created it is checked against this schema. http://json-schema.org/.
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      required: ['farm_id', 'name'],
+      properties: {
+        id: { type: 'integer' },
+        farm_id: { type: 'string' },
+        name: { type: 'string' },
+        notes: { type: ['string', 'null'] },
+        ...this.baseProperties,
+      },
+      additionalProperties: false,
+    };
+  }
```

Comment:
Could possibly add the relationMappings here for the relationship tables to avoid inserting multiple models and just upsert using the graph instead. I think both are acceptable just wondering the preference.

Reply:
Hmmm, I'm interested in that! It seems weird to have models with no associated controllers + routes, but I hate the (un-) readability of the `.withGraphFetched()` syntax. Do you necessarily have to construct one of those graph strings in order to take advantage of `relationMappings`? Do you have an existing example you like?

Reply:
My understanding is that whole point of the relationMapping is to use graph-like queries/insertions. So you dont have to use it at all but if you want to receive or insert data in one call you can. I will try to use it in the batch requests and see if it is liked pattern.

Reply:
I added the relationMappings and it made the logic for the GET method a lot easier, thanks for that! Not sure how it works from the insert aspect though